client/my-sites/upgrades/components/form/state-select.jsx

```diff
@@ -47,38 +43,37 @@ module.exports = React.createClass( {
 			return (
 				<Input { ...this.props } />
 			);
-		} else {
-			options.push( { key: '', label: this.translate( 'Select State' ), disabled: 'disabled' } );
+		}
 
-			options = options.concat( statesList.map( function( state ) {
-				if ( isEmpty( state.code ) ) {
-					return { key: '--', label: '', disabled: 'disabled' };
-				} else {
-					return { key: state.code, label: state.name };
-				}
-			} ) );
+		options.push( { key: '', label: this.translate( 'Select State' ), disabled: 'disabled' } );
 
-			return (
-				<div className={ classes }>
-					<div>
-						<FormLabel htmlFor={ this.props.name }>{ this.props.label }</FormLabel>
-						<FormSelect
-							ref="input"
-							name={ this.props.name }
-							value={ this.props.value }
-							disabled={ this.props.disabled }
-							onBlur={ this.handleBlur }
-							onChange={ this.props.onChange }
-							onClick={ this.recordStateSelectClick }
-							onFocus={ this.handleFocus } >
-						{ options.map( function( option ) {
-							return <option key={ option.key } value={ option.key } disabled={ option.disabled }>{ option.label }</option>;
-						} ) }
-						</FormSelect>
-					</div>
-					{ this.props.errorMessage && <FormInputValidation text={ this.props.errorMessage } isError /> }
+		options = options.concat( statesList.map( function( state ) {
+			if ( isEmpty( state.code ) ) {
```

Comment:
I know it's not introduced here but `isEmpty` is mostly for empty arrays and objects with no own properties. If we know that `state.code` is a string, we can just make a truthy check `!! state.code` 

> A value is considered empty if it’s an arguments object, array, string, or jQuery-like collection with a length of 0 or has no own enumerable properties.